x-pack/test/alerting_api_integration/common/fixtures/plugins/alerts/server/alert_types.ts

```diff
@@ -286,6 +286,50 @@ export function defineAlertTypes(
     },
     async executor(opts: AlertExecutorOptions) {},
   };
+  const throwAlertType: AlertType = {
+    id: 'test.throw',
+    name: 'Test: Throw',
+    actionGroups: [
+      {
+        id: 'default',
+        name: 'Default',
+      },
+    ],
+    producer: 'alerting',
+    defaultActionGroupId: 'default',
+    async executor({ services, params, state }: AlertExecutorOptions) {
+      throw new Error('this alert is intended to fail');
+    },
+  };
+  const patternFiringAlertType: AlertType = {
+    id: 'test.patternFiring',
+    name: 'Test: Firing on a Pattern',
+    actionGroups: [{ id: 'default', name: 'Default' }],
+    producer: 'alerting',
+    defaultActionGroupId: 'default',
+    async executor(alertExecutorOptions: AlertExecutorOptions) {
+      const { services, state, params } = alertExecutorOptions;
+      const pattern = params.pattern;
+      if (!Array.isArray(pattern)) throw new Error('pattern is not an array');
+      if (pattern.length === 0) throw new Error('pattern is empty');
+
+      // get the pattern index, return if past it
+      const patternIndex = state.patternIndex ?? 0;
+      if (patternIndex > pattern.length) {
+        return { patternIndex };
+      }
+
+      // fire if pattern says to
+      if (pattern[patternIndex]) {
+        services.alertInstanceFactory('instance').scheduleActions('default');
+      }
+
+      return {
+        patternIndex: (patternIndex + 1) % pattern.length,
+      };
+    },
+  };
```

Comment:
Nah.  This alert takes a "pattern" of booleans that indicate if it should fire on specific executions.  Eg, [false, true, false, true] means on the 1s and 3rd executions it will schedule actions, but on the 2nd and 4th it will.  I realized I was going to need something like this to be able to test the new-instance / resolved-instance events, in the getAlertStatus() PR, so I did a first pass here (getAlertStatus() will require per-instance patterns, or something :-) ).